solidity/test/TestQuerySender.t.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.13;
+
+import "forge-std/Test.sol";
+import {InterchainQueryRouter} from "../contracts/middleware/InterchainQueryRouter.sol";
+import {TestQuerySender} from "../contracts/test/TestQuerySender.sol";
+import {HyperlaneTestEnvironment} from "./HyperlaneTestEnvironment.sol";
+
+import {MockToken} from "../contracts/mock/MockToken.sol";
+
+contract TestQuerySenderTest is Test {
+    HyperlaneTestEnvironment testEnvironment;
+    TestQuerySender sender;
+    uint32 originDomain = 123;
+    uint32 destinationDomain = 321;
+
+    function setUp() public {
+        testEnvironment = new HyperlaneTestEnvironment(
+            originDomain,
+            destinationDomain
+        );
+
+        sender = new TestQuerySender();
+        sender.initialize(address(testEnvironment.queryRouters(originDomain)));
+    }
+
+    function testSendAddressQuery(address owner) public {
+        vm.assume(owner != address(0x0));
```

Comment:
this prevents `transferOwnership` revert?

Reply:
Yes